From ac5c5a3b6d704045744c64e43b9c5e0d034004c6 Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Thu, 1 Feb 2024 11:53:18 +0530
Subject: [PATCH] crypto: disable self-test by default for low mem profiles.

It can now be enabled using bootargs by setting
cryptomgr.notests=0

This patch is snapshot of the commit
a9919a662ef487234a3ac7941f107e159aff6689
from win.linuxopenwrt.2.1 branch and
oss/kernel/linux-ipq-5.4 project

Change-Id: Ida67b3106e93624b416fa3d0a66588e265531ada
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
---
 crypto/testmgr.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/crypto/testmgr.c b/crypto/testmgr.c
index 7aaaefb653df..82a4a8c0fdaf 100644
--- a/crypto/testmgr.c
+++ b/crypto/testmgr.c
@@ -40,7 +40,12 @@
 
 MODULE_IMPORT_NS(CRYPTO_INTERNAL);
 
+#if CONFIG_IPQ_MEM_PROFILE == 256 || CONFIG_IPQ_MEM_PROFILE == 512
+static bool notests=1;
+#else
 static bool notests;
+#endif
+
 module_param(notests, bool, 0644);
 MODULE_PARM_DESC(notests, "disable crypto self-tests");
 
-- 
2.34.1

